### PR TITLE
Add contributing guide

### PR DESCRIPTION
**Pull request by kytrinyx**
This is essentially a straight copy from the documentation, but
I updated the list of available languages.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+Log it in our [issue tracker][ghi] or send a
+note to the [mailing list][ml].
+Be sure to include all relevant information, like the versions of Sinatra and
+Ruby you're using. A [gist](http://gist.github.com/) of the code that caused
+the issue as well as any error messages are also very helpful.
+
+## Need help?
+
+The [Sinatra mailing list][https://groups.google.com/forum/#!forum/sinatrarb] has
```
Comment (kgrz) [inline CONTRIBUTING.md]: Should  the link here be in parentheses `()`? GitHub's rich-diff renderer doesn't seem to parse this properly.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+   if the patch is doc related.
+
+2. **Write unit tests:** If you add or modify functionality, it must
+   include unit tests. If you don't write tests, we have to, and this
+   can hold up acceptance of the patch.
+
+3. **Mind the `README`:** If the patch adds or modifies a major feature,
+   modify the `README.md` file to reflect that. Again, if you don't
+   update the `README`, we have to, and this holds up acceptance.
+
```
Comment (kgrz) [inline CONTRIBUTING.md]: Should we include here an instruction to update the Changelog as necessary?

File: CONTRIBUTING.md (hunk 1 above)
Comment (kytrinyx) [inline CONTRIBUTING.md]: Yeah that looks wrong. Let me check the old page that I ported as well.

File: CONTRIBUTING.md (hunk 2 above)
Comment (kytrinyx) [inline CONTRIBUTING.md]: That would be handy. I'll go ahead and add that.

File: CONTRIBUTING.md (hunk 1 above)
Comment (kytrinyx) [inline CONTRIBUTING.md]: My bad. I converted it from the old [ml] notation to the full link, before I realized how the [ml] thing worked. I'm switching that back now.

File: CONTRIBUTING.md (hunk 2 above)
Comment (kytrinyx) [inline CONTRIBUTING.md]: Should we add a section to the top of the changelog?

```
= Next Release

 * Your change here
```

That would let people know where to put their change.

Comment (kytrinyx) [inline CONTRIBUTING.md]: Update -- that change did go into the contributing guide, so I can reference it from here.

Comment (kytrinyx) [inline CONTRIBUTING.md]: I've added a step about the Changelog. I'd welcome suggestions for improvements to the phrasing here.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+4. **Update the Changelog (`CHANGES`):** The changelog helps give an
+   overview of the changes that go into each release, and gives credit
+   where credit is due. We make sure that the Changelog is up to date
+   before each release, and we always appreciate it when people make
+   it easier to get the release out the door.
```
Comment (kytrinyx) [inline CONTRIBUTING.md]: I've added a step about the Changelog. I'd welcome suggestions for improvements to the phrasing here.

**Pull request merged**